freeipa/src/main/java/com/sequenceiq/freeipa/client/ClusterProxyErrorRpcListener.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.freeipa.client;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.googlecode.jsonrpc4j.JsonRpcClient;
+import com.sequenceiq.cloudbreak.clusterproxy.ClusterProxyException;
+
+import java.util.Optional;
+
+public class ClusterProxyErrorRpcListener implements JsonRpcClient.RequestListener {
+
+    private static Optional<ClusterProxyError> deserializeAsClusteProxyError(ObjectNode objectNode) {
+        ObjectMapper mapper = new ObjectMapper();
+        ClusterProxyError clusterProxyError;
+        try {
+            clusterProxyError = mapper.treeToValue(objectNode, ClusterProxyError.class);
+            if (!clusterProxyError.code.contains("cluster-proxy")) {
+                return Optional.empty();
+            }
+        } catch (Exception ex) {
+            return Optional.empty();
+        }
+        return Optional.of(clusterProxyError);
+    }
+
+    @VisibleForTesting
```

Comment:
the logic should be tested using public methods if possible. I think in this case it's easy to test via `onBeforeResponseProcessed`

Reply:
true, I'll move it to the public method.